java/src/main/java/genericpubsub/ManagedSubscribe.java

```diff
@@ -58,13 +54,29 @@ public void startManagedSubscription() {
         serverStream = asyncStub.managedSubscribe(this);
         ManagedFetchRequest.Builder builder = ManagedFetchRequest.newBuilder().setNumRequested(BATCH_SIZE);
 
+        if (Objects.nonNull(managedSubscriptionId)) {
+            builder.setSubscriptionId(managedSubscriptionId);
+            logger.info("Starting managed subscription with ID {}", managedSubscriptionId);
+        } else if (Objects.nonNull(developerName)) {
+            builder.setDeveloperName(developerName);
+            logger.info("Starting managed subscription with developer name {}", developerName);
+        } else {
+            logger.warn("No ID or developer name specified");
+        }
+
         if (Objects.nonNull(developerName)) {
             builder.setDeveloperName(developerName);
         }
         if (Objects.nonNull(managedSubscriptionId)) {
             builder.setSubscriptionId(managedSubscriptionId);
         }
         serverStream.onNext(builder.build());
+
+        // Thread being blocked here for demonstration of this specific example. Blocking the thread in production is not recommended.
+        while(isActive.get()) {
```

Comment:
This is another change based on what we already have in the [Subscribe](https://github.com/forcedotcom/pub-sub-api/blob/main/java/src/main/java/genericpubsub/Subscribe.java#L84-L87) example